pkg/api/preflight.go

```diff
@@ -0,0 +1,67 @@
+package api
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"encoding/json"
+)
+
+// PreflightRequest is the request body of preflight
+type PreflightRequest struct {
+	Resources []json.RawMessage `json:"resources"`
+}
+
+// ValidationResult is the validation result to return in the deployment preflight response body
+type ValidationResult struct {
+	Status ValidationStatus            `json:"status"`
+	Error  *ManagementErrorWithDetails `json:"error,omitempty"`
+}
+
+type ManagementErrorWithDetails struct {
+	// Code - The error code returned from the server.
+	Code *string `json:"code,omitempty"`
+	// Message - The error message returned from the server.
+	Message *string `json:"message,omitempty"`
+	// Target - The target of the error.
+	Target *string `json:"target,omitempty"`
+	// Details - Validation error.
+	Details *[]ManagementErrorWithDetails `json:"details,omitempty"`
+}
+
+func (m *ManagementErrorWithDetails) String() string {
```

Comment:
can you add a simple unit test?

Reply:
I could do that, but I also might just remove it since I'm not currently using it. Will check aks again